src/pkg/packager/create.go

```diff
@@ -90,6 +93,50 @@ func (p *Packager) Create() (err error) {
 		return err
 	}
 
+	// copy cosign tree logic to build image list of signatures
+	var nameOpts []name.Option
+
+	// Add signatures and attestations for images
+	if p.cfg.CreateOpts.SkipCosignLookup {
+		message.Debug("Skipping cosign artifact lookups per --skip-cosign-lookup flag")
+	} else {
+		for i, c := range p.cfg.Pkg.Components {
+			var imageSigList []string
+			for _, image := range c.Images {
+				ref, err := name.ParseReference(image, nameOpts...)
+
+				if err != nil {
+					return err
+				}
+
+				var remoteOpts []ociremote.Option
+				simg, _ := ociremote.SignedEntity(ref, remoteOpts...)
+				if simg == nil {
+					continue
+				}
+				sigRef, _ := ociremote.SignatureTag(ref, remoteOpts...)
+				attRef, _ := ociremote.AttestationTag(ref, remoteOpts...)
+
+				sigs, err := simg.Signatures()
+				if err == nil {
+					layers, _ := sigs.Layers()
+					if len(layers) > 0 {
+						imageSigList = append(imageSigList, sigRef.String())
+					}
+				}
+
+				atts, err := simg.Attestations()
+				if err == nil {
+					layers, _ := atts.Layers()
+					if len(layers) > 0 {
+						imageSigList = append(imageSigList, attRef.String())
+					}
+				}
+			}
+			p.cfg.Pkg.Components[i].Images = append(p.cfg.Pkg.Components[i].Images, imageSigList...)
+		}
+	}
```

Comment:
This should be in a helper (and probably really 2 helpers).

- The first that deals with looping through images should be a receiver on `Packager` that handles the control flow of adding the signatures and attestations to the package.
- The second should be in `src/utils/sget` (which should be renamed `src/utils/cosign` to do the signature and attestations for a given image reference.

Reply:
I think I got this adjusted in the latest commits - I wasn't sure where to place the first one so I just left it in create for now.